ocaml/xapi/records.ml

```diff
@@ -1997,3 +1997,89 @@ let sdn_controller_record rpc session_id sdn_controller =
         make_field ~name:"port"                ~get:(fun () -> Int64.to_string (x ()).API.sDN_controller_port) ();
       ]}
 
+let pusb_record rpc session_id pusb =
+  let _ref = ref pusb in
+  let empty_record = ToGet (fun () -> Client.PUSB.get_record rpc session_id !_ref) in
+  let record = ref empty_record in
+  let x () = lzy_get record in
+  { setref=(fun r -> _ref := r; record := empty_record );
+    setrefrec=(fun (a,b) -> _ref := a; record := Got b);
+    record=x;
+    getref=(fun () -> !_ref);
+    fields =
+      [
+        make_field ~name:"uuid" ~get:(fun () -> (x ()).API.pUSB_uuid) ();
+        make_field ~name:"usb-group-uuid"
+        ~get:(fun () -> try get_uuid_from_ref (x ()).API.pUSB_USB_group with _ -> nid) ();
+        make_field ~name:"host-uuid" ~get:(fun () -> get_uuid_from_ref (x ()).API.pUSB_host) ();
+        make_field ~name:"host-name-label" ~get:(fun () -> try get_name_from_ref (x ()).API.pUSB_host with _ -> nid) ();
+        make_field ~name:"path" ~get:(fun () -> (x ()).API.pUSB_path) ();
+        make_field ~name:"vendor_id" ~get:(fun () -> (x ()).API.pUSB_vendor_id) ();
+        make_field ~name:"vendor_desc" ~get:(fun () -> (x ()).API.pUSB_vendor_desc) ();
```

Comment:
Inconsistent hyphens and underscores here.

Reply:
ok. should be vendor-id and vendor-desc

Reply:
There are a lot of other instances of this further down in the file too.

Reply:
updated